### ISSUE TITLE
January is not recognized by date_extractor?

### ISSUE THREAD

**Issue by sherbondy** [spencermountain/nlp_compromise#15]
Heya,

This is a wonderful library. Hoping to use it to extract dates in a project, but I noticed that January consistently fails to be properly extracted in tests. I'm wondering if this is a subtle bug with indexes / accidental type coercion of 0 to false in date_extractor.coffee.

Would be happy to help you track down the issue if you have trouble.

Here is an example I just tried on master:
```
nlp.value("Today is January 7, 2015").date()
{ month: null,
  day: 7,
  year: 2015,
  to_day: null,
  to_year: 2015,
  to_month: null }
```

**Comment by sherbondy** [comment]
Hey, you know what, I think I figured it out:

parseInt("0") || null
returns null

Dat rough JavaScript lyfe.

**Comment by sherbondy** [comment]
How about we replace || with ? and call it a day? Gonna submit a pull request in a little bit after I do some tests.

**Comment by spencermountain** [comment]
Oh Ethan, you're a dear. Will merge any fix, thanks for the nice words

**Comment by sherbondy** [comment]
Well, I tried to fix things, but something went awry.
https://github.com/inboxapp/nlp_compromise

January works, but now half the time I call date(), null is returned.
There seems to be some bit of state that is persisting between calls, which is quite unfortunate.
Let me know if you get to the bottom of this, or manage to successfully cherry-pick a subset of these changes into a working fix!

**Comment by spencermountain** [comment]
hey man, this is fixed in v0.3.0
